clients/src/main/java/org/apache/kafka/common/network/KafkaChannel.java

```diff
@@ -236,6 +241,27 @@ public ChannelMuteState muteState() {
         return muteState;
     }
 
+    /**
+     * Mute this channel on {@link #prepare()} failure. This will remove all operations from this channel.
+     */
+    private void muteOnPrepareFailure() {
```

Comment:
It would be better to use `AuthenticationFailure` rather than `PrepareFailure` here and in other methods.